File: src/main/java/net/siegemc/core/items/Rarity.java
package net.siegemc.core.items;

public enum Rarity {
    COMMON("Common", "7"),
    UNCOMMON("Uncommon", "8"),
    RARE("Rare", "9"),
    EPIC("Epic", "5"),
    LEGENDARY("Legendary", "6"),
    DEBUG("Debug", "c");
    
    private final String color;
    private final String id;
    
    Rarity(String id, String string) {
        this.color = string;
        this.id = id;
    }
    
    public String getColor() {
        return this.color;
    }
    
    public String getID() {
        return this.id;
    }
}

File: src/main/kotlin/net/siegemc/core/items/Rarity.kt
package net.siegemc.core.items

enum class Rarity(val id: String, val color: String) {
    COMMON("Common", "<gray>"),
    UNCOMMON("Uncommon", "<green>"),
    RARE("Rare", "<blue>"),
    EPIC("Epic", "<dark_purple>"),
    LEGENDARY("Legendary", "<gold>"),
    SPECIAL("Special", "<rainbow>"),
    DEBUG("Debug", "<red>");

    companion object {
        fun getFromId(id: String?): Rarity {
            for (rarity in values()) {
                if (rarity.id.equals(id, ignoreCase = true)) return rarity
            }
            return COMMON
        }

        fun getFromInt(int: Int): Rarity {
            return when (int) {
                in 0..39 -> COMMON
                in 40..59 -> UNCOMMON
                in 60..79 -> RARE
                in 80..94 -> EPIC
                in 95..100 -> LEGENDARY
                in 101..150 -> SPECIAL
                else -> DEBUG
            }
        }
    }
}